Allow user scaling on mobile devices

Alex-Jordan opened the pull request.

rbeezer commented:
Thanks for moving on this one, and for actually moving this one.  Two things:

(1) I want to check that you have experimented with this on a small device, since I think it must have been done for some reason.

(2) I'd like David F to check-off, since this was built as a part of the UI CSS/JS.

Alex-Jordan commented:
I checked on my Android phone with Chrome. Now I can zoom with my fingers,
and I can't do that using the mathbook.pugestsound sample chapter.

I can zoom on images in addition to being able to tap them to make them
scale to 100%. I think that is fine and good.

At:
spot.pcc.edu/~ajordan/temp
there is the sample article produced on this branch. Test it on your phones.


On Fri, May 25, 2018 at 7:48 PM, Rob Beezer <notifications@github.com>
wrote:

> Thanks for moving on this one, and for actually moving this one. Two
> things:
>
> (1) I want to check that you have experimented with this on a small
> device, since I think it must have been done for some reason.
>
> (2) I'd like David F to check-off, since this was built as a part of the
> UI CSS/JS.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/rbeezer/mathbook/pull/889#issuecomment-392230440>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AEg3AJzp6Pf8QDWBCIrmpFZV3zMXxpiTks5t2MJ3gaJpZM4UOv-0>
> .
>



-- 
Alex Jordan
Mathematics Instructor
Portland Community College

rbeezer commented:
Chrome/Android.  Works well - 4-button panel nav panel at the bottom zooms in 
size, but also moves outside the visible screen.

Firefox/Android.  Content zooms fine.  4-button nav panel at the bottom stays in 
place, zoom makes it taller, then presumably widths adjust and the buttons get 
tall and skinny.  At max-zoom, buttons occupy half the screen.  Not unusable, 
but not very professional.

iPhone.  Don't have one.  Not sure if iPad is good stand-in.  Anyway, we could 
build a beta and ask -dev to test?  Or try to add some more to this to make 
Firefox behavior better first?

On 05/25/2018 08:22 PM, Rob Beezer wrote:
> Very good. Doors closing.
> 
> On May 25, 2018 8:11:54 PM PDT, Alex Jordan <notifications@github.com> wrote:
>  >I checked on my Android phone with Chrome. Now I can zoom with my
>  >fingers,
>  >and I can't do that using the mathbook.pugestsound sample chapter.
>  >
>  >I can zoom on images in addition to being able to tap them to make them
>  >scale to 100%. I think that is fine and good.
>  >
>  >At:
>  >spot.pcc.edu/~ajordan/temp
>  >there is the sample article produced on this branch. Test it on your
>  >phones.
>  >
>  >
>  >On Fri, May 25, 2018 at 7:48 PM, Rob Beezer <notifications@github.com>
>  >wrote:
>  >
>  >> Thanks for moving on this one, and for actually moving this one. Two
>  >> things:
>  >>
>  >> (1) I want to check that you have experimented with this on a small
>  >> device, since I think it must have been done for some reason.
>  >>
>  >> (2) I'd like David F to check-off, since this was built as a part of
>  >the
>  >> UI CSS/JS.
>  >>
>  >> —
>  >> You are receiving this because you authored the thread.
>  >> Reply to this email directly, view it on GitHub
>  >>
>  ><https://github.com/rbeezer/mathbook/pull/889#issuecomment-392230440>,
>  >or mute
>  >> the thread
>  >>
>  ><https://github.com/notifications/unsubscribe-auth/AEg3AJzp6Pf8QDWBCIrmpFZV3zMXxpiTks5t2MJ3gaJpZM4UOv-0>
>  >> .
>  >>
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/rbeezer/mathbook/pull/889#issuecomment-392232407>, or mute 
> the thread 
> <https://github.com/notifications/unsubscribe-auth/ABy2cp0lJFnb-yntd2FCB83PKMV3_p0Hks5t2MqHgaJpZM4UOv-0>.
>

rbeezer commented:
Any more interest in this one?  I always ahve aplogize at JMM when people try a two-finger zoom.  The Firefox/Android behavior is definitely a drawback - it'd be nice to isolate that problem, fix it,  and maybe get an overall better setup in the process.

rbeezer commented:
New version going up now at

https://mathbook.pugetsound.edu/beta/20180926-zoom/derivatives.html

I'll test shortly and report.

Alex-Jordan commented:
If you report the OS/browser/etc that you test on, I can try to supplement by tapping some colleagues here in my department.

Alex-Jordan commented:
Two years later :)

I manually applied this change to this page:
http://spot.pcc.edu/~ajordan/temp/graphics.html

Just tested on all of the following:

- Android Chrome
- Android Firefox
- iPhone Safari
- iPhone Chrome

And it behaves in all four like Rob described for Android Chrome:

> Works well - 4-button panel nav panel at the bottom zooms in
size, but also moves outside the visible screen.

The place it wasn't behaving before was Firefox. I hadn't used FF on my phone in probably two years, and at first it was still bad. Then I updated FF, and it works. Seems reasonable this was a FF bug before.

Fingers crossed I can cross off another old PR. If you do look at merging this, note that I removed two extra space characters, but for some reason not the 3rd. You knock that one off too at merge time.

Alex-Jordan commented:
Oh, and there is a "conflict" now, but it appears to just be that one of these lines is immediately adjacent to the relatively newer favicon line.

I could close and make a new PR. You will likely merge something separate anyway, so let me know if that helps.

rbeezer commented:
Beta for crowdsourced testing is out on -dev.  Works well for me in Firefox/Android on my tablet.  Fingers crossed.

https://pretextbook.org/beta/2020-05-13-zoom
